core/app/beyond/metrics/SystemMetricsWriter.scala

```diff
@@ -72,38 +74,18 @@ object SystemMetricsWriter {
     (JsPath \ "date").write[Date] and
     (JsPath \ "count").write[Int]
   )(unlift(NumberOfRequestsPerSecond.unapply)).addTag("NumberOfRequestsPerSecond")
-}
-
-class SystemMetricsWriter extends Actor with ActorLogging with MongoMixin {
-
-  import SystemMetricsWriter._
-  import play.api.Play.current
-  import play.api.libs.concurrent.Akka
 
-  implicit val ec: ExecutionContext = Akka.system.dispatcher
-
-  private def collection: JSONCollection = db.collection[JSONCollection]("admin.metrics")
-
-  override def preStart() {
-    log.info("SystemMetricsWriter started")
-  }
-
-  override def postStop() {
-    log.info("SystemMetricsWriter stopped")
-  }
-
-  // FIXME: Needs error handling.
-  override def receive: Receive = {
-    case msg: SystemLoadAverage =>
-      collection.save(Json.toJson(msg))
-    case msg: HeapMemoryUsage =>
-      collection.save(Json.toJson(msg))
-    case msg: NonHeapMemoryUsage =>
-      collection.save(Json.toJson(msg))
-    case msg: SwapMemoryUsage =>
-      collection.save(Json.toJson(msg))
-    case msg: NumberOfRequestsPerSecond =>
-      collection.save(Json.toJson(msg))
-  }
+  private def collection: JSONCollection =
```

Comment:
To avoid potential problems with Play hot-reloading. [FYI](https://github.com/ReactiveMongo/Play-ReactiveMongo/blob/master/samples/Play-ReactiveMongo-Sample/app/controllers/UsingJsonReadersWriters.scala#L39-L41)
Beyond uses def for collection until now.